pkg/controller/utils/utils.go

```diff
@@ -104,7 +106,47 @@ func DiscoveryFromPodTemplate(w *unstructured.Unstructured, fields ...string) ([
 	return ports, spec.Labels, nil
 }
 
-func getContainerPorts(cs []v1.Container) []intstr.IntOrString {
+// DiscoveryHelmModuleWorkload will get the workload created by flux/helm-controller
+func DiscoveryHelmModuleWorkload(ctx context.Context, c client.Reader, comp *v1alpha2.Component, ns string) (*unstructured.Unstructured, error) {
+	if comp == nil || comp.Spec.Helm == nil {
+		return nil, errors.New("the component has no valid helm module")
+	}
+
+	rls, err := util.RawExtension2Unstructured(&comp.Spec.Helm.Release)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get helm release from component")
+	}
+	rlsName := rls.GetName()
+
+	chartName, _, err := unstructured.NestedString(rls.Object, fluxapis.HelmChartNamePath...)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get helm chart name")
+	}
+
+	// qualifiedFullName is used as the name of target workload.
+	// It strictly follows the convention that Helm generate default full name as below:
+	// > We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+	// > If release name contains chart name it will be used as a full name.
+	qualifiedWorkloadName := rlsName
+	if !strings.Contains(rlsName, chartName) {
+		qualifiedWorkloadName = fmt.Sprintf("%s-%s", rlsName, chartName)
+		if len(qualifiedWorkloadName) > 63 {
+			qualifiedWorkloadName = strings.TrimSuffix(qualifiedWorkloadName[:63], "-")
+		}
+	}
+
+	wl, err := util.RawExtension2Unstructured(&comp.Spec.Workload)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get workload from component")
+	}
+
+	if err := c.Get(ctx, client.ObjectKey{Namespace: ns, Name: qualifiedWorkloadName}, wl); err != nil {
```

Comment:
1. Besides the release name, is there any other possible name?
2. We should check the workload annotation(helm release annotation), and make sure it's created by helm and match the release

Reply:
1. I think we can make it extensible to adopt other possible naming conventions.  The current one we used is a recommended practice by Helm official doc. Maybe we can listen to the feedback from community.
> It is highly recommended that new charts are created via helm create command as the template names are automatically defined as per this best practice.
2. Good idea.